llama-rs/src/lib.rs

```diff
@@ -1505,15 +1499,19 @@ impl InferenceSession {
         prompt: &str,
         maximum_token_count: Option<usize>,
         rng: &mut impl rand::Rng,
-        callback: impl Fn(&str) -> Result<(), E>,
+        mut callback: impl FnMut(&str) -> Result<(), E>,
```

Comment:
:+1:  Nice catch. It's always good API design to make callbacks FnMut when we don't have special immutability requirements on our end